OneSignalDevelopers/ngosi
# Prework to send notifications to attendees

Fixes #56 

## Solution

~I'm utilizing Supabase's Realtime feature to subscribe to events on the `Survey` table events.~

I added the plumbing needed to send notification emails to a presenter's audience using OneSignal.
* `setEmailDevice` - creates the device record in OneSignal required to send emails to a user.
* `updateTags` - submit data tags to OneSignal to be injected into an email.
* `sendEmail` - sends an email to the given email.

These ☝🏾 functions require the `ONESIGNAL_API_KEY` to be set in `.env.local`.

The preso endpoint, `api/preso`, was updated to handle `PUT` requests to enable presenters to update their audience with the video to their published talk. Submitting a request to this endpoint will result in an email being sent to each attendee.

## Testing

These steps assume you already have a preso created and an app configured for email on OneSignal.

1. Select an existing preso
2. Navigate to the survey for the press by changing the URL from `localhost:3000/preso/<id>` to `localhost:3000/survey/<id>`
3. Note that the survey page is for audience-users
4. Fill out the survey a few times using different email addresses and notification settings (note down the addresses that should receive an email)
5. Navigate to the preso details page
6. Insert a value for the _Published content location_ field and click the **Save** button
7. Check the email address you noted down for emails 

## Resources

### References

- ~[Managing Realtime](https://supabase.com/docs/guides/api#managing-realtime)~
- ~[Subscribing to events](https://supabase.com/docs/reference/javascript/subscribe)~
